src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -76,8 +76,15 @@ create_topic_properties_update(
     update.properties.segment_size.op = op_t::remove;
     update.properties.timestamp_type.op = op_t::remove;
     update.properties.retention_bytes.op = op_t::remove;
+    update.properties.remote_delete.op = op_t::remove;
+    update.properties.segment_ms.op = op_t::remove;
     update.properties.shadow_indexing.op = op_t::remove;
     update.properties.retention_duration.op = op_t::remove;
+    update.properties.batch_max_bytes.op = op_t::remove;
+    update.properties.retention_local_target_ms.op = op_t::remove;
+    update.properties.retention_local_target_bytes.op = op_t::remove;
+    update.properties.initial_retention_local_target_ms.op = op_t::remove;
+    update.properties.initial_retention_local_target_bytes.op = op_t::remove;
```

Comment:
same for these

Reply:
will ping product

Reply:
> will ping product

Did we get an update here?

Reply:
Noah said he pinged @mattschumpert to give his feedback on what we want to do here, so we're waiting for him to chime in on this.

Reply:
@dotnwat I am wondering if you are concerned about some of the configs in particular (eg. `shadow_indexing_mode`) or the change in general (making AlterConfigs work like AlterConfigs is supposed to and not like IncrementalAlterConfigs).

For the shadow indexing mode configs in particular I am wondering if we should make AlterConfig reset `shadow_indexing_mode` and `remote_delete` to their cluster-level defaults (but not std::nullopt) to match the behaviour in CreateTopic. Something like this: https://github.com/redpanda-data/redpanda/commit/889c48d594388570f5da11a04ef463f7daf3716d. Although it might even be preferred to not touch these two configs unless they are specified but I can see downsides to both.